src/TLuaInterpreter.cpp

```diff
@@ -2305,7 +2305,7 @@ int TLuaInterpreter::setFontSize(lua_State* L)
         lua_pushfstring(L, "setFontSize: bad argument #%d type (size as number expected, got %s!)", s, luaL_typename(L, s));
         return lua_error(L);
     } else {
-        size = lua_tonumber(L, s);
+        size = lua_tointeger(L, s);
```

Comment:
💭 The clang-code-model always gives a warning about assignments using `lua_tointeger`.

For the record: `lua_tointeger()` returns a `lua_Integer` and this seems to be a `ptrdiff_t` on my amd64 Linux OS (so is a _64-bit signed integer_ type) - however `int` (and indeed `long`) on a Windows platform is only 32 bits - to get a big enough value to silence the warning (_the yellow triangle in the margin if you have the clang-code-model enabled - sadly it is a resource 🐖 and will happily consume several GBytes of memory and a process core for a lot of the time - so you may not want to have it switched on by default_) you will want to either use `lua_Integer` for `size` or use:

    size = static_cast<int>(lua_tointeger(L, s));